City-of-Helsinki/drupal-helfi-platform-config
# UHF-3268 News item

How to test / install:
- Make sure your instance is up and running on latest dev branch, preferably Strategia & Talous instance.
- Update the modules/themes
   - `composer require drupal/hdbt:dev-UHF-3268_news_item && composer require drupal/helfi_platform_config:dev-UHF-3268_news_item && composer require drupal/hdbt_admin:dev-UHF-3268_news_item`
- Go to shell `make shell`
  - In shell, type `drush en -y helfi_news_item`
  - Exit shell
- Run updates, import translations and clear caches `make drush-updb drush-locale-update drush-cr`
- Log in to the site and go to create few news tags: https://strategia.docker.so/fi/paatoksenteko-ja-hallinto/admin/structure/taxonomy/manage/news_tags/add
   - No need to add anything else than just the Name field and hit <enter>
   - Hox! The news tags are themed, but not visible until the news tag pages are built
- Go to create news item(s) https://strategia.docker.so/fi/paatoksenteko-ja-hallinto/node/add/news_item
   - Tip: Try copying real content from https://www.hel.fi/uutiset/fi 
   - Add the following fields:  Title (Otsikko), Lead-in (Johdanto), Main image (Pääkuva), Content (Pääsisältö), Links (Linkit) and select News tags (Uutistagit) and from sidebar Short title (Lyhyt otsikko)
- After you have enough content, test that: 
   - Publish and unpublish works (disappears from the latest news list)
   - Translations works (only current language content is listed)
   - Short title works (shows short title in lists, url and breadcrumb but not in the news item itself)
- Check the specs from https://helsinkisolutionoffice.atlassian.net/browse/UHF-3268 as the task itself is already done.
- This PR also has the theming included. Check that the layout looks and works as designed: https://share.goabstract.com/1d0a8fe1-a4e6-4160-8a5d-95e3c66b7cf1?collectionLayerId=d329ae36-624f-4a7a-b75c-ef37f2194376&mode=design&sha=latest   
   

**Bonus**
- Explore the site and inspect images. Check that the images have width and height attributes set, like so:
![image](https://user-images.githubusercontent.com/1712902/146542756-fd4d1ec7-28da-4d92-8439-06abc25023e7.png)

Check also: 
- https://github.com/City-of-Helsinki/drupal-hdbt/pull/210
- https://github.com/City-of-Helsinki/drupal-hdbt-admin/pull/98
